ml_dtypes/tests/custom_float_test.py

```diff
@@ -830,6 +830,41 @@ def testDivmod(self, float_type):
         float_type=float_type,
     )
 
+  @ignore_warning(category=RuntimeWarning, message="invalid value encountered")
+  @ignore_warning(category=RuntimeWarning, message="divide by zero encountered")
+  def testDivmodZeroDenominator(self, float_type):
+    x = np.array([-np.inf, -1, 0, np.nan, 1, np.inf], dtype=float_type)
+    y = np.array(0.0, dtype=float_type)
+    o1, o2 = np.divmod(x, y)
+    e1, e2 = np.divmod(x.astype(np.float32), y.astype(np.float32))
+    numpy_assert_allclose(
```

Comment:
the following still fails locally
```py
import ml_dtypes
import numpy as np
from numpy import float16, float32
from ml_dtypes import bfloat16, float8_e5m2, float8_e4m3

xv=[0.0, 1.0, "inf", "nan", -0.0, -1.0, "-inf", "-nan"]
yv=[0.0, 1.0, "inf", "nan", -0.0, -1.0, "-inf", "-nan"]
resv_f16 = [np.floor_divide(float16(x), float16(y)) for x in xv for y in yv]
resv_f32 = [np.floor_divide(float32(x), float32(y)) for x in xv for y in yv]
## OK f16 vs f32
np.testing.assert_allclose(resv_f16, resv_f32)

resv_bf16 = [np.floor_divide(bfloat16(x), bfloat16(y)) for x in xv for y in yv]
## FAILED f16 vs bf16
np.testing.assert_allclose(resv_f16, resv_bf16)
```

Reply:
The discrepancy is from ml_dtypes not treating `-0.0` as different than `0.0`:
```python
In [3]: np.divmod(ml_dtypes.bfloat16('1.0'), ml_dtypes.bfloat16('-0.0'))
Out[3]: (inf, nan)

In [4]: np.divmod(np.float16('1.0'), np.float16('-0.0'))
Out[4]: (np.float16(-inf), np.float16(nan))
```
It's not entirely clear to me how important this is. Do you know whether this is part of the IEEE spec?

Reply:
It is important. According to the IEEE 754 standard for floating-point arithmetic, the result of dividing a positive number by negative zero (e.g., 1.0 / -0.0) is negative infinity (-∞).

This follows the rule that division by zero leads to infinity, with the sign determined by the signs of the operands.

Reply:
`floor_divide` mismatches found for the following operands

`np.floor_divide(a,b)`

| a    | b | float16 | bfloat16  | 
| -------- | ------- | -------- | -------- |
| 1.0  | -0.0    | -inf | inf |
| inf  | -0.0    | -inf | inf |
| -1.0 | -0.0    | inf | -inf |
| -inf    | -0.0    | inf | -inf |

```
op  a  b  expected actual

<ufunc 'floor_divide'> 1.0 -0.0 -inf inf
<ufunc 'floor_divide'> inf -0.0 -inf inf
<ufunc 'floor_divide'> -1.0 -0.0 inf -inf
<ufunc 'floor_divide'> -inf -0.0 inf -inf
```

Reply:
Thanks - I've updated the implementation to handle signed zeros in the denominator in the same way NumPy does.